public/plugin.ts

```diff
@@ -27,24 +27,30 @@ import {
   ClientConfigType,
 } from './types';
 import { LOGIN_PAGE_URI, PLUGIN_NAME } from '../common';
+import { API_ENDPOINT_PERMISSIONS_INFO } from './apps/configuration/constants';
 
 export class OpendistroSecurityPlugin
   implements Plugin<OpendistroSecurityPluginSetup, OpendistroSecurityPluginStart> {
   // @ts-ignore : initializerContext not used
   constructor(private readonly initializerContext: PluginInitializerContext) {}
 
-  public setup(core: CoreSetup): OpendistroSecurityPluginSetup {
+  public async setup(core: CoreSetup) {
```

Comment:
Please keep the return type

Reply:
Originally I intentionally removed it as the async annotation requires the return type to be wrapped up with a promise. However, at that time, there were some issues with my dev branch where the plugin failed to start. But now I have tested it again with the return type and it is working fine.